src/layout/dom.ts

```diff
@@ -141,6 +167,9 @@ export function update(node: Node, data: NodeInfo, source: Source): void {
             }
         }
 
+        // If element has a valid shadowRoot, track Shadow DOM as a top level root node.
+        if ("shadowRoot" in element && element.shadowRoot && !has(element.shadowRoot)) { discoverTree(element.shadowRoot); }
```

Comment:
Very minor; Might move this logic to a separate function to avoid repeating it a few places.

Reply:
I simplified this so it won't look as bad.